src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDashboard/requestLog.tsx

```diff
@@ -0,0 +1,329 @@
+import {t} from 'app/locale';
+import {SentryApp, SentryAppWebhookRequest, SentryAppSchemaIssueLink} from 'app/types';
+
+const ALL_EVENTS = t('All Events');
+const MAX_PER_PAGE = 10;
+
+const componentHasSelectUri = (issueLinkComponent: SentryAppSchemaIssueLink): boolean => {
+  const hasSelectUri = (fields: any[]): boolean =>
+    fields.some(field => field.type === 'select' && 'uri' in field);
+
+  const createHasSelectUri =
+    hasSelectUri(issueLinkComponent.create.required_fields) ||
+    hasSelectUri(issueLinkComponent.create.optional_fields || []);
+
+  const linkHasSelectUri =
+    hasSelectUri(issueLinkComponent.link.required_fields) ||
+    hasSelectUri(issueLinkComponent.link.optional_fields || []);
+
+  return createHasSelectUri || linkHasSelectUri;
+};
+
+const getEventTypes = memoize((app: SentryApp) => {
+  // TODO(nola): ideally this would be kept in sync with EXTENDED_VALID_EVENTS on the backend
+
+  let issueLinkEvents: string[] = [];
+  const issueLinkComponent = (app.schema.elements || []).find(
+    element => element.type === 'issue-link'
+  );
+  if (issueLinkComponent) {
+    issueLinkEvents = ['external_issue.created', 'external_issue.linked'];
+    if (componentHasSelectUri(issueLinkComponent as SentryAppSchemaIssueLink)) {
+      issueLinkEvents.push('select_options.requested');
+    }
+  }
+
+  const events = [
+    ALL_EVENTS,
+    'installation.created',
+    'installation.deleted',
+    ...(app.events.includes('error') ? ['error.created'] : []),
+    ...(app.events.includes('issue')
+      ? ['issue.created', 'issue.resolved', 'issue.ignored', 'issue.assigned']
+      : []),
+    ...(app.isAlertable ? ['event_alert.triggered'] : []),
+    ...issueLinkEvents,
+  ];
+
+  return events;
+});
+
+const ResponseCode = ({code}: {code: number}) => {
+  let priority = 'error';
+  if (code <= 399 && code >= 300) {
+    priority = 'warning';
+  } else if (code <= 299 && code >= 100) {
+    priority = 'success';
+  }
+
+  return (
+    <div>
+      <Tag priority={priority}>{code === 0 ? 'timeout' : code}</Tag>
+    </div>
+  );
+};
+
+const TimestampLink = ({date, link}: {date: moment.MomentInput; link?: string}) => {
+  return link ? (
+    <Link to={link}>
+      <DateTime date={date} />
+    </Link>
+  ) : (
+    <DateTime date={date} />
+  );
+};
+
+type Props = AsyncComponent['props'] & {
+  app: SentryApp;
+};
+
+type State = AsyncComponent['state'] & {
+  eventType: string;
+  errorsOnly: boolean;
+  currentPage: number;
+  requests: SentryAppWebhookRequest[];
+};
+
+export default class RequestLog extends AsyncComponent<Props, State> {
+  shouldReload = true;
+
+  get hasNextPage() {
+    return (this.state.currentPage + 1) * MAX_PER_PAGE < this.state.requests.length;
+  }
+
+  get hasPrevPage() {
+    return this.state.currentPage > 0;
+  }
+
+  getEndpoints(): Array<[string, string, any] | [string, string]> {
+    const {slug} = this.props.app;
+
+    const query: any = {};
+    if (this.state) {
```

Comment:
@billyvg not here because this gets called in the constructor of AsyncComponent

Reply:
Ah, initialization of `this.state` depends on this method. Not needed for this PR, but I think it'd be better UX if these states were stored in the URL vs local state.

Reply:
In the future maybe we can move the request log to a separate URL (using NavTabs), and then it would definitely make sense to store these in the query params